### PR TITLE
Improve consistency by using response from PUT requests directly to save state

### PR DESCRIPTION
**Pull request by zippolyte**
Avoid consistency issues, due to replication delays, caused by calling the GET endpoint right after PUT.
This leaves the Create method as it is, as there is already a retry on 404 for this problem, and we need this for updating dashboard lists when used.

Also fixes some linting warnings

### CODE REVIEW

**Review by matt-miller-ddog — DISMISSED** [review]

File: datadog/resource_datadog_dashboard.go [hunk 1]
```diff
@@ -193,7 +198,7 @@ func updateDashboardLists(d *schema.ResourceData, providerConf *ProviderConfigur
 	}
 }
 
-func loadDatadogDashboard(d *schema.ResourceData, dashboard datadogV1.Dashboard) error {
+func loadDatadogDashboard(d *schema.ResourceData, dashboard *datadogV1.Dashboard) error {
```
Comment (nmuesch) [inline datadog/resource_datadog_dashboard.go:201]: Small nit, can we name this how you're doing the newer ones; i.e. monitor is going to have - `updateMonitorState`

**Review by nmuesch — APPROVED** [review]
LGTM, approving but left one nit

**Review by nmuesch — APPROVED** [review]

**Pull request merged**